spring-kafka/src/main/java/org/springframework/kafka/aot/KafkaRuntimeHintsRegistrar.java

```diff
@@ -0,0 +1,239 @@
+					ContainerProperties.class,
+					ProducerListener.class)
+				.forEach(type -> reflectionHints.registerType(type,
+						builder -> builder.withMembers(MemberCategory.INVOKE_DECLARED_METHODS)));
+
+		Stream.of(
+					Message.class,
+					ImplicitLinkedHashCollection.Element.class,
+					NewTopic.class,
+					AbstractKafkaListenerContainerFactory.class,
+					ConcurrentKafkaListenerContainerFactory.class,
+					KafkaListenerContainerFactory.class,
+					KafkaListenerEndpointRegistry.class,
+					DefaultKafkaConsumerFactory.class,
+					DefaultKafkaProducerFactory.class,
+					KafkaAdmin.class,
+					KafkaOperations.class,
+					KafkaResourceFactory.class,
+					KafkaTemplate.class,
+					ProducerFactory.class,
+					KafkaOperations.class,
+					ConsumerFactory.class,
+					LoggingProducerListener.class,
+					ImplicitLinkedHashCollection.Element.class,
+					KafkaListenerAnnotationBeanPostProcessor.class)
+				.forEach(type -> reflectionHints.registerType(type,
+						builder -> builder.withMembers(MemberCategory.INVOKE_DECLARED_CONSTRUCTORS,
+								MemberCategory.INVOKE_DECLARED_METHODS,
+								MemberCategory.INTROSPECT_PUBLIC_METHODS)));
+
+		Stream.of(
+					KafkaBootstrapConfiguration.class,
+					CreatableTopic.class,
+					KafkaListenerEndpointRegistry.class)
+				.forEach(type -> reflectionHints.registerType(type,
+						builder -> builder.withMembers(MemberCategory.INVOKE_DECLARED_CONSTRUCTORS)));
+
+		Stream.of(
+					AppInfo.class,
+					// standard assignors
+					CooperativeStickyAssignor.class,
+					RangeAssignor.class,
+					RoundRobinAssignor.class,
+					StickyAssignor.class,
+					// standard partitioners
+					DefaultPartitioner.class,
+					RoundRobinPartitioner.class,
+					UniformStickyPartitioner.class,
+					// standard serialization
+					ByteArrayDeserializer.class,
+					ByteArraySerializer.class,
+					ByteBufferDeserializer.class,
+					ByteBufferSerializer.class,
+					BytesDeserializer.class,
+					BytesSerializer.class,
+					DoubleSerializer.class,
+					DoubleDeserializer.class,
+					FloatSerializer.class,
+					FloatDeserializer.class,
+					IntegerSerializer.class,
+					IntegerDeserializer.class,
+					ListDeserializer.class,
+					ListSerializer.class,
+					LongSerializer.class,
+					LongDeserializer.class,
+					StringDeserializer.class,
+					StringSerializer.class,
+					// Spring serialization
+					DelegatingByTopicDeserializer.class,
+					DelegatingByTypeSerializer.class,
+					DelegatingDeserializer.class,
+					ErrorHandlingDeserializer.class,
+					DelegatingSerializer.class,
+					JsonDeserializer.class,
+					JsonSerializer.class,
+					ParseStringDeserializer.class,
+					StringOrBytesSerializer.class,
+					ToStringSerializer.class)
+				.forEach(type -> reflectionHints.registerType(type, builder ->
+						builder.withMembers(MemberCategory.INVOKE_PUBLIC_CONSTRUCTORS)));
+
+		reflectionHints.registerType(TypeReference.of("java.util.zip.CRC32C"), builder ->
```

Comment:
Why FQCN? Look like this class is a part of Java by itself

Reply:
Again; copied from spring-native.